### ISSUE TITLE
on mobile app under android, can't open or download OpenDocument

### ISSUE THREAD

**Issue by bcag2** [mattermost/mattermost-mobile#5873]
I saw similar issue as https://forum.mattermost.org/t/can-not-download-attachment-on-mobile-app/12342, so I upgrade my mattermost client from 1.46 to 1.48 with no changes,
I saw also https://github.com/mattermost/mattermost-mobile/issues/4809, seems a little different.

#### Summary
I can open pdf file with OpenDocument reader app, **but I can't open OpenDocument file** (.odt in my case), and I can't download my .odt file.

### Environment Information
  - Device Name: Samsung Galaxy A7 (2018) 64 Go
  - OS Version: Androïd 10
  - Mattermost App Version: 1.48.0
  - Mattermost Server Version: 5.39.2

#### Steps to reproduce
Share an opendocument (.odt in my case) from desktop client for example, then try to read it in mobile client.
In my case, I installed OpenDocument Reader v3.10.4 (https://f-droid.org/en/packages/at.tomtasche.reader/) 
For a pdf, mattermost suggest to open the file :  
![Screenshot_20211212-095832_Android System](https://user-images.githubusercontent.com/5690299/145707872-6b7cfa39-cc08-41f9-b95e-91cf5decfd6c.jpg)

Now, on my .odt file (you can show it on previous capture), mattermost suggest only to download it, then show this screen :  
![Screenshot_20211212-095901_Android System](https://user-images.githubusercontent.com/5690299/145707920-58481386-dad2-4ddc-973b-065e3bec574b.jpg)
 and I can't launch download 

#### Possible fixes
"just" allow open with the same app … OpenDocument reader for OpenDocument as for pdf file !-)

**Comment by amyblais** [comment]
@josephbaylon Are you able to reproduce this on Android?

**Comment by bcag2** [comment]
Yes, I can. Do you need additional screenshot?

**Comment by twnk** [comment]
Having an issue where trying to open a PDF on Android redirects to the Google Play Store with “application/pdf” prefilled in the search bar.

**Comment by twnk** [comment]
Warning from https://www.npmjs.com/package/react-native-file-viewer — I can't see in the code where file extensions are specified during the file viewing, I wonder if these being missing are causing Android to reject the files as non-openable? 

```typescript
// *IMPORTANT*: The correct file extension is always required.
// You might encounter issues if the file's extension isn't included
// or if it doesn't match the mime type of the file.
// https://stackoverflow.com/a/47767860
```

**Comment by Pheester** [comment]
I am having the same issue as twnk along with word and excel.

**Comment by amyblais** [comment]
> Having an issue where trying to open a PDF on Android redirects to the Google Play Store with “application/pdf” prefilled in the search bar.

We have released a v1.49.1 dot release to fix this issue - should be available in the app stores within 1 or 2 days.

**Comment by bcag2** [comment]
Thanks a lot, I upgraded to v1.49 and it works fine now
